File: Examples/DotNET/SDK/CSharp/Manipulate/PerformSeveralOperationsOnImage.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Imaging. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////

using System;
using Aspose.Cloud;

namespace Aspose.Imaging.Cloud.Examples.Manipulate
{
    class PerformSeveralOperationsOnImage
    {
        static void Main() 
        {
            string dataDir = Common.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
            string input = "sample1.jpg";
            string output = "output.jpg";

            string outPath ="Imaging/"+input;

            int newWidth = 100;
            int newHeight = 100;
            int x = 5;
            int y = 5;
            int rectWidth = 10;
            int rectHeight = 10;
            
            Common.StorageService.File.UploadFile(dataDir + input, input, storage: Common.STORAGE);

            Common.GetImagingSdk().UpdateImage.PerformScalingCroppingAndFlippingOfAnImage(input, 
                ImageFormat.Jpg, newWidth, newHeight, x, y, rectWidth, rectHeight, outPath, Common.FOLDER, storage: Common.STORAGE);

            Common.StorageService.File.DownloadFile(outPath, dataDir + output, storage: Common.STORAGE);
            
        }
    }
}

File: Examples/Python/SDK/manipulate/PerformSeveralOperationsOnImage.py
import common

input_file = "sample1.png"
input_path = common.get_path(__file__, input_file)

output_file = "output.jpg"
output_path = common.get_path(__file__, output_file)

format = "jpg"
x = 96
y = 96
newWidth = 300
newHeight = 300
rectWidth = 200
rectHeight = 200
rotateFlipMethod = ""
# invoke Aspose.Imaging Cloud SDK API to perform format change operation without using cloud storage
response = common.imagingApi.PostImageSaveAs_ImagingApi_0(format, newWidth, newHeight, x, y, rectWidth, rectHeight, rotateFlipMethod, file=input_path)

if response.Status == 'OK':
    # download image from API response
    with open(output_path, 'wb') as f:
        for chunk in response.InputStream:
            f.write(chunk)
